### PR TITLE
PMM-7304 minikube operator setup script

### PR DESCRIPTION
**Pull request by puneet0191**
Following: https://github.com/percona/pmm-doc/blob/master/docs/setting-up/server/dbaas.md

### CODE REVIEW

File: pmm-tests/minikube_operators_setup.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+PMM_USER_B64="$(echo -n "${PMM_USER}" | base64)";
+PMM_PASS_B64="$(echo -n "${PMM_PASS}" | base64)";
+
+# Install the PXC operator
+curl -sSf -m 30 https://raw.githubusercontent.com/percona/percona-xtradb-cluster-operator/pmm-branch/deploy/bundle.yaml | minikube kubectl -- apply -f -
+curl -sSf -m 30 https://raw.githubusercontent.com/percona/percona-xtradb-cluster-operator/pmm-branch/deploy/secrets.yaml | sed "s/pmmserver:.*=/pmmserver: ${PMM_PASS_B64}/g" | minikube kubectl -- apply -f -
+
+# Install the PSMDB operator
+curl -sSf -m 30 https://raw.githubusercontent.com/percona/percona-server-mongodb-operator/v1.6.0/deploy/bundle.yaml | minikube kubectl -- apply -f -
```
Comment (atymchuk) [inline pmm-tests/minikube_operators_setup.sh:15]: Should the version (v1.6.0) be saved to a variable for the purpose of re-using/changing it later?

Comment (puneet0191) [reply pmm-tests/minikube_operators_setup.sh:15]: you have valid point, we should change it variable, I will make the required changes and push here.

**Comment by atymchuk** [comment]
@puneet0191 This whole script has little to do with tests, it belongs in the Jenkins pipeline. Even more so since the other part fo the setup routine is actually there, i.e. the installation of docker, minikube and so on (https://github.com/Percona-Lab/jenkins-pipelines/pull/738/files#). This feels like an organic part of that pipeline.

File: pmm-tests/minikube_operators_setup.sh (hunk 1 above)
Comment (puneet0191) [reply pmm-tests/minikube_operators_setup.sh]: I have fixed this, and the pipeline now takes the operator version as a parameter

File: pmm-tests/minikube_operators_setup.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+# Prepare a set of base64 encoded values and non encoded for user and pass with administrator privileges to pmm-server (DBaaS)
+PMM_USER='admin';
+PMM_PASS='admin';
+OPERATOR_VERSION=$1;
```
Comment (atymchuk) [inline pmm-tests/minikube_operators_setup.sh:6]: For the sake of better DX I suggest we check if the variable is actually passed, and error out if not:
`if [ -z "$OPERATOR_VERSION" ] ; then
    echo "Error: no parameter passed"
    echo "Use: bash ./minikube_operators_setup.sh <operator_version>"
    exit 1
fi
`

File: pmm-tests/minikube_operators_setup.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+PMM_USER_B64="$(echo -n "${PMM_USER}" | base64)";
+PMM_PASS_B64="$(echo -n "${PMM_PASS}" | base64)";
+
+# Install the PXC operator
+curl -sSf -m 30 https://raw.githubusercontent.com/percona/percona-xtradb-cluster-operator/pmm-branch/deploy/bundle.yaml | minikube kubectl -- apply -f -
+curl -sSf -m 30 https://raw.githubusercontent.com/percona/percona-xtradb-cluster-operator/pmm-branch/deploy/secrets.yaml | sed "s/pmmserver:.*=/pmmserver: ${PMM_PASS_B64}/g" | minikube kubectl -- apply -f -
+
+# Install the PSMDB operator
+curl -sSf -m 30 https://raw.githubusercontent.com/percona/percona-server-mongodb-operator/v${OPERATOR_VERSION}/deploy/bundle.yaml | minikube kubectl -- apply -f -
+curl -sSf -m 30 https://raw.githubusercontent.com/percona/percona-server-mongodb-operator/v${OPERATOR_VERSION}/deploy/secrets.yaml | sed "s/PMM_SERVER_USER:.*$/PMM_SERVER_USER: ${PMM_USER}/g;s/PMM_SERVER_PASSWORD:.*$/PMM_SERVER_PASSWORD: ${PMM_PASS}/g;" | minikube kubectl -- apply -f -
```
Comment (atymchuk) [inline pmm-tests/minikube_operators_setup.sh:17]: Can we wrap this line?

File: pmm-tests/minikube_operators_setup.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+PMM_USER='admin';
+PMM_PASS='admin';
+OPERATOR_VERSION=$1;
+
+PMM_USER_B64="$(echo -n "${PMM_USER}" | base64)";
+PMM_PASS_B64="$(echo -n "${PMM_PASS}" | base64)";
+
+# Install the PXC operator
+curl -sSf -m 30 https://raw.githubusercontent.com/percona/percona-xtradb-cluster-operator/pmm-branch/deploy/bundle.yaml | minikube kubectl -- apply -f -
+curl -sSf -m 30 https://raw.githubusercontent.com/percona/percona-xtradb-cluster-operator/pmm-branch/deploy/secrets.yaml | sed "s/pmmserver:.*=/pmmserver: ${PMM_PASS_B64}/g" | minikube kubectl -- apply -f -
```
Comment (atymchuk) [inline pmm-tests/minikube_operators_setup.sh:13]: Consider wrapping this line for readability

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: pmm-tests/minikube_operators_setup.sh (hunk 2 above)
Comment (puneet0191) [reply pmm-tests/minikube_operators_setup.sh:6]: fixed

File: pmm-tests/minikube_operators_setup.sh (hunk 4 above)
Comment (puneet0191) [reply pmm-tests/minikube_operators_setup.sh]: yes done

File: pmm-tests/minikube_operators_setup.sh (hunk 3 above)
Comment (puneet0191) [reply pmm-tests/minikube_operators_setup.sh]: Done

**Review by atymchuk — APPROVED** [review]

**Pull request merged**